components/konflux-ui/staging/base/proxy/proxy.yaml

```diff
@@ -52,8 +52,59 @@ spec:
           requests:
             cpu: 10m
             memory: 64Mi
+      - name: render-runtime-config
+        image: registry.access.redhat.com/ubi9/ubi@sha256:66233eebd72bb5baa25190d4f55e1dc3fff3a9b77186c1f91a0abdb274452072
+        envFrom:
+        - secretRef:
+            name: watson-config
+        command:
+          - sh
+          - -c
+          - |
+            set -e
+
+            # Generate runtime-config.js with PUBLIC_ variables
+            out=/mnt/static-content/runtime-config.js
+            printf "window.KONFLUX_RUNTIME = window.KONFLUX_RUNTIME || {};\n" > "$out"
+            env | grep -E '^PUBLIC_' | while IFS='=' read -r name value; do
+              esc=$(printf '%s' "$value" | sed -e 's/\\\\/\\\\\\\\/g' -e 's/\"/\\\"/g')
+              printf 'window.KONFLUX_RUNTIME.%s = "%s";\n' "$name" "$esc" >> "$out"
+            done
+
+            # Apply cluster-specific variables from ConfigMap files to runtime-config.js, if present
+            if [ -d /mnt/ui-runtime-env-variables ]; then
```

Comment:
I would prefer to keep the secret with only secret data and to store the rest of the data in the konflux-info config map.